papers/ring_vrf/pedersen_vrf.tex

```diff
@@ -99,52 +112,75 @@ \section{Ring VRF construction}% {Ring VRFs from the Pedersen VRF}
+	\item $\rVRF.\KeyGen $ is the same as $ \PedVRF.\KeyGen$.
+	\item $\rVRF.\Eval $ is the same as $ \PedVRF.\Eval$.
+	\item $ \rVRF.\CommitRing: \pkring \rightarrow \comring$ given a set of public keys $ \pkring $, it returns a Merkle tree root where the leaves are elements of  $ \pkring$. 
+	
+	We note that our construction works fine by replacing  $ \comring $ by $ \pkring $. However, the asymptotic complexity of our protocol with $ \comring $ is much more better
+	%COMMENT Handan: We can improve the previous sentence.
+	
+	%COMMENT Handan: We don't need to define opring here because our construction does not need it.
 \end{itemize}
 
+%COMMENT Handan: It is better to talk about these stuffs after describing the protocol.
+%As described in \S\ref{sec:overview},
+%we instantiate a rVRF-AD from \PedVRF plus a ring commitment scheme
+% $\rVRF.\{ \CommitRing, \OpenRing \}$.
+%In our construction, $\rVRF.\Eval = \PedVRF.\Eval$.
+%and $\rVRF.\KeyGen = \PedVRF.\KeyGen$.
+%We alter $\rVRF.\KeyGen$ somewhat in \S\ref{subsec:rvrf_faster}
+%by adopting a SNARK friendly public key,
+% although the \pk here still exists under the hood.
+%We discuss further nuances of this in \S\ref{subsec:AML_KYC}.
+
+	As mentioned in \S\ref{sec:overview}, the signing algorithm of our ring VRF construction shows that the public key of the committed secret key is in the ring. Therefore, we need a zero-knowledge ring membership proof which is provided by $ \rVRF.\Sign $ that we define below.
+%	 for the relation \Lring
+%	which handles both $\PedVRF.\OpenKey$ and $\rVRF.\OpenKey$ efficiently.
+%	% \vspace{-0.1in}
+%	$$ \Lring = \Setst{ \compk, \comring }{
+%		\eprint{ \exists \openpk,\openring \textrm{\ s.t.\ } }{}
+%		\genfrac{}{}{0pt}{}{\PedVRF.\OpenKey(\compk,\openpk) \quad}{\,\, = \rVRF.\OpenRing(\comring,\openring)}
+%	} \mathperiod $$
+%	
+	
+	\def\tmpaux{\aux \doubleplus \compk \doubleplus \piring \doubleplus \comring}
+	\def\tmpeprintaux{\eprint{\aux'}{\tmpaux}}
+	\def\tmpindent{\hspace*{5pt}}
+	\begin{itemize}
+		\item $\rVRF.\rSign\eprint{ : }{}(\sk,\msg,\aux)\eprint{ \mapsto \rho}{}$ returns $\rho = (\compk,\piring,\sigma)$ \eprint{where \\}{\\ where} 
+		\tmpindent $(\openpk,\compk) \leftarrow \PedVRF.\CommitKey(\sk)$,  \\
+		\tmpindent $\piring \leftarrow \NIZK_{\mathcal{R}_{ring}}.\Prove((\compk,\comring); \openpk)$, \\
+		where 
+		\begin{align}
+				\mathcal{R}_{ring} = \{(\openpk;(\genG,\genB,\grE,\ring, \compk)): \\(\compk-\openpk \, \genB)  \in \ring\} \label{rel:snark} \nonumber
+			\end{align}
+		\tmpindent $\aux' \leftarrow \tmpaux$,  \\
+		\tmpindent $\sigma \leftarrow \PedVRF.\Sign(\sk,\openpk,\msg, \aux')$.
+		\item $\rVRF.\rVerify : (\comring,\msg,\aux,\rho) \mapsto \Out \,\, \lor \perp$ \,
+		parses $\rho$ as $(\compk,\piring,\sigma)$, \eprint{next }{}sets $\aux' \leftarrow \tmpaux$,
+		aborts if $\NIZK_{\Lring}.\Verify((\compk,\comring); \piring)$ fails,
+		and returns $\PedVRF.\Verify(\compk,\msg, \aux', \sigma)$.
+	\end{itemize}
+	
+
+
+
```

Comment:
On line 198:
- I would replace "our ring VRF construction" with "our ring VRF instantiation". 
- Instead of "the zero-knowledge property of $ \NIZK_{\Lring}$" should be "the zero-knowledge property of $ \NIZK_{\Rring}$"
- H' replaced by H (BTW, H' not used?); H_G' replaced by H_G.
- $\sigma_1$ and  $ \sigma_1$ I believe should look like
$ \sigma_1 = (\PreOut_1,\compk, \pi_{\PedVRF}) $ and $ \sigma_2 = (\PreOut_2, \compk, \pi'_{\PedVRF})
On  line 201: do you rather mean "assuming that NIZK_Reval and NIZK_Rring  are zero-knowledge and knowledge sound".

Reply:
I think we should say construction here because we describe a generic protocol with some cryptographic primitives having some security properties. Instantiation makes more sense for example when we can instantiate our construction with the zk SNARK described in Section 5.